extensions/shared/paid-blocks/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { Fragment } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { BlockControls } from '@wordpress/block-editor';
+import { ToolbarGroup, Button } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getUpgradeUrl, isUpgradable } from "./utils";
+
+const jetpackPaidBlockEdit = createHigherOrderComponent(
```

Comment:
How would you feel about separating this component out and letting `index.js` import it and set the filter?